### PR TITLE
Support for invoking a function value

### LINKED ISSUE #95: Invoke a v8go.Value ?
Hi,
If I have a v8go.Value where `IsFunction()` returns `true`, is there any way to invoke that value? 
Thank you!
Rob

### PR DESCRIPTION
**Pull request by robfig**
Fixes #95

### CODE REVIEW

**Comment by robfig** [comment]
This is a work in progress -- still need to add more tests -- but let me know if I'm not on the right track.

Thanks!

**Comment by robfig** [comment]
OK, this should have 100% test coverage. The current failure is about a missing FOSSA key(?)

**Review by tmc — APPROVED** [review]
This all looks reasonable to me. @rogchap ?

File: function.go
Comment (rogchap) [inline function.go:19]: Can we change the `args` to be the `Valuer` interface? Then you can use any Value type. ie Object etc

File: function.go
Comment (rogchap) [inline function.go:23]: Is this else statement needed? The range would still work with a empty array?
If we need to pass the nil pointer can we return in the above `if` statement to avoid the extra branching caused by the `else`

**Review by rogchap — CHANGES_REQUESTED** [review]
This looks great. Thank you for putting this PR together, just a few minor comments/changes

File: function.go
Comment (tmc) [reply function.go:19]: Good catch!

File: function.go
Comment (robfig) [reply function.go]: That was something I found by writing the test -- to get the `*C.ValuePtr`, I have to take the value of the first element, and that is an illegal index on an empty slice. 

Updated to avoid calling FunctionCall in two different branches.

File: function.go
Comment (robfig) [reply function.go]: Updated

**Comment by robfig** [comment]
Thanks for the review. Comments addressed, and tests pass locally.

File: function_test.go
Comment (rogchap) [inline function_test.go:28]: This is looking great @robfig.
Sorry to be a pain, but looking at this I'm wondering if a variadic function would be more idiomatic? Then you could do:
```
fn.Call(arg1, arg2, etc)
```

**Review by rogchap — CHANGES_REQUESTED** [review]
I should have mentioned this in the first review, sorry for the back and forth. Please could you also update the `CHANGELOG.md`? Don't forget to add your attribution 💪

File: function_test.go
Comment (robfig) [reply function_test.go]: No problem at all -- agreed that it looks cleaner. Updated

**Comment by robfig** [comment]
Added & updated!

**Comment by rogchap** [comment]
Perfect @robfig 
Thanks again. I'll fix up the FOSSA error (GitHub secrets don't work in forks)

**Review by rogchap — APPROVED** [review]

**Comment by robfig** [comment]
Sounds good -- not sure if you expect me to merge, but I don't have access to do that despite your approval. I'll leave it to you.

**Comment by rogchap** [comment]
I'll merge shortly.

**Pull request merged**
